daemon/graphdriver/driver.go

```diff
@@ -173,9 +199,19 @@ func New(root string, options []string) (driver Driver, err error) {
 	return nil, fmt.Errorf("No supported storage backend found")
 }
 
+func scanPriorDrivers(root string) []string {
+	priorDrivers := []string{}
```

Comment:
Can this be a `map[string]struct{}` instead so we don't have to double-loop above?